cli/benchmark.py

```diff
@@ -68,22 +68,20 @@ def run(
 
     \b
     <benchmark_id> is formatted like BENCHMARK_NAME==BENCHMARK_VERSION.
-    <agent_config> is the path to an agent configuration file.
+    <agent_locator> is the locator string for the registered agent.
```

Comment:
We have `_raise_on_invalid_locator(locator)` function guarding the `ClassRegister.make()`. So perhaps we don't have to check locator validity in other parts of the code?

https://github.com/huawei-noah/SMARTS/blob/86d7b4dbce092395e27123a92a35ff18e7d646b6/smarts/core/utils/class_factory.py#L130-L134
https://github.com/huawei-noah/SMARTS/blob/86d7b4dbce092395e27123a92a35ff18e7d646b6/smarts/core/utils/class_factory.py#L106-L108

Reply:
Fair enough, I was mentioning this since it is best practice to exit as early as possible. The factory side check waits all the way to the point where we call `registry.make`. That is after installing dependencies, reading configuration, and uncovering the environment.

The check is probably better to go through the registry interface instead:

https://github.com/Farama-Foundation/Gymnasium/blob/76559ed553cffc3e5082227b0a8acea461273e84/gymnasium/envs/registration.py#L115-L136

Reply:
I just thought we could hack it using `NAME_CONSTRAINT_REGEX` but that is perhaps inappropriate to use the regex object directly in case of changing the underlying implementation.